x/epoching/types/query.go

```diff
@@ -18,12 +20,39 @@ func (e *Epoch) ToResponse() *EpochResponse {
 
 // ToResponse parses a QueuedMessage into a query response queued message struct.
 func (q *QueuedMessage) ToResponse() *QueuedMessageResponse {
+	var enriched *EnrichedMsg
+	sdkMsg := q.UnwrapToSdkMsg()
+
+	switch msg := sdkMsg.(type) {
```

Comment:
I think we need to recover type for every possible message that can queued, not only `MsgDelegate` and `MsgDelegate`